### PR TITLE
Fix spurious fail log messages

### PR DESCRIPTION
**Pull request by jumaffre**
Resolves https://github.com/microsoft/CCF/issues/387. 

Connections between the host and the enclave are now properly terminated, even when the TLS client terminates the connection gracefully. Note that when the TCP socket is invalidated, no more messages are written to and from the enclave.

Sequence diagram of new behaviour:
image

Note that when the TCP connection is closed from the host, no more messages are written to the enclave (there might still be messages to write out to the peer) and the following logic applies:

image

### CODE REVIEW

**Comment by cimetrics** [comment]
images

File: src/host/rpcconnections.h [hunk 1]
```diff
@@ -172,6 +171,16 @@ namespace asynchost
       return s->second->write(len, data);
     }
 
+    bool stop(int64_t id)
+    {
+      // Invalidating the TCP socket will result in the handle being close. No
```
Comment (eddyashton) [inline src/host/rpcconnections.h]: Typo: Should be "being close**d**".

And "read from" isn't necessarily true - between this point and the enclave processing the `tls_close` message, additional messages about this `id` may be written by the enclave and read by the host.

Comment (eddyashton) [reply src/host/rpcconnections.h]: So we need a null-check in `RPCConnections::write`, as `s->second` may be `nullptr`.

**Comment by cimetrics** [comment]
images

**Pull request merged**